crates/kubelet/src/node.rs

```diff
@@ -444,16 +387,233 @@ fn node_labels_definition(arch: &str, config: &Config) -> HashMap<String, String
+        status: &str,
+        timestamp: &DateTime<Utc>,
+        reason: &str,
+        message: &str,
+    ) {
+        self.conditions
+            .push(k8s_openapi::api::core::v1::NodeCondition {
+                type_: type_.to_string(),
+                status: status.to_string(),
+                last_heartbeat_time: Some(Time(timestamp.clone())),
+                last_transition_time: Some(Time(timestamp.clone())),
+                reason: Some(reason.to_string()),
+                message: Some(message.to_string()),
+            });
+    }
+
+    /// Add a address to the node.
+    pub fn add_address(&mut self, type_: &str, address: &str) {
+        self.addresses
+            .push(k8s_openapi::api::core::v1::NodeAddress {
+                type_: type_.to_string(),
+                address: address.to_string(),
+            });
+    }
+
+    /// Build node definition from builder.
+    pub fn build(self) -> Node {
+        let mut metadata: k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta =
+            Default::default();
+        metadata.name = Some(self.name);
+        metadata.annotations = Some(self.annotations);
+        metadata.labels = Some(self.labels);
+
+        let mut spec: k8s_openapi::api::core::v1::NodeSpec = Default::default();
+        spec.pod_cidr = Some(self.pod_cidr);
+        spec.taints = Some(self.taints);
+
+        let mut node_info: k8s_openapi::api::core::v1::NodeSystemInfo = Default::default();
+        node_info.architecture = self.architecture;
+        node_info.kube_proxy_version = self.kube_proxy_version;
+        node_info.kubelet_version = self.kubelet_version;
+        node_info.container_runtime_version = self.container_runtime_version;
+        node_info.operating_system = self.operating_system;
+
+        let mut status: k8s_openapi::api::core::v1::NodeStatus = Default::default();
+        status.node_info = Some(node_info);
+        status.capacity = Some(self.capacity);
+        status.allocatable = Some(self.allocatable);
+        status.daemon_endpoints = Some(k8s_openapi::api::core::v1::NodeDaemonEndpoints {
+            kubelet_endpoint: Some(k8s_openapi::api::core::v1::DaemonEndpoint { port: self.port }),
+        });
+        status.conditions = Some(self.conditions);
+        status.addresses = Some(self.addresses);
+
+        let kube_node = k8s_openapi::api::core::v1::Node {
+            metadata: Some(metadata),
+            spec: Some(spec),
+            status: Some(status),
+        };
+        Node(kube_node)
+    }
+}
+
+impl Default for NodeBuilder {
+    fn default() -> NodeBuilder {
+        NodeBuilder {
+            name: "krustlet".to_string(),
+            annotations: BTreeMap::new(),
+            labels: BTreeMap::new(),
+            pod_cidr: "10.244.0.0/24".to_string(),
+            taints: vec![],
+            architecture: "".to_string(),
+            kube_proxy_version: "v1.17.0".to_string(),
+            // TODO Should this be the crate version?
```

Comment:
Yeah, this should probably be the crate version

Reply:
Cool